substreams-helper/src/math.rs

```diff
@@ -26,17 +26,12 @@ pub fn decimal_from_hex_be_bytes(price_bytes: &Vec<u8>) -> BigDecimal {
         .with_prec(100);
 }
 
-pub fn exponent_to_big_decimal(decimals: &BigInt) -> BigDecimal {
+pub fn exponent_to_big_decimal(decimals: u8) -> BigDecimal {
```

Comment:
Why modify the old one? We probably want both

Reply:
Bc we are not calculating 10^bigint. The function is only for 10^token_decimals afaik

Reply:
I understand we are not using it here, but we might need to use it elsewhere. (This code snippet is copied from the Uniswap v3 substream and they use `BigInt` pretty extensively throughout their codebase)

Reply:
most of the time, `decimals` is from `decimals: u64` of `Erc20Token`.

- first of all, I feel it'd be better to just do `decimals: u8` - that's already 0-255, large enough for decimals
- with `exponent_to_big_decimal(decimals: u8)` we don't need to do the unit -> bigint conversion right?

but anyway if you want to keep both, I can keep both, but will that make someone misuse them?

Reply:
yeah, that sounds good. Let's keep decimals u8 then